src/JLD.jl

```diff
@@ -749,9 +749,9 @@ length(x::Union(JldFile, JldGroup)) = length(names(x))
 
 is_valid_type_ex(s::Symbol) = true
 is_valid_type_ex(s::QuoteNode) = true
-is_valid_type_ex(x::Int) = true
+is_valid_type_ex{T}(::T) = isbits(T)
 is_valid_type_ex(e::Expr) = ((e.head == :curly || e.head == :tuple || e.head == :.) && all(map(is_valid_type_ex, e.args))) ||
-                            (e.head == :call && (e.args[1] == :Union || e.args[1] == :TypeVar))
+                            (e.head == :call && isa(e.args[1], Symbol))
```

Comment:
It's a little unfortunate that we can't be more specific here, but there's no way to determine if calling a symbol might end up as a bitstype immutable without trying it.